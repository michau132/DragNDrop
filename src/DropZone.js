import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Paper, Typography } from '@material-ui/core';

const HeightPaper = styled(Paper)`
  && {
    margin-top: 20px;
    height: 250px;
    width: 90%;
    border: 2px dashed ${({ dragging }) => (dragging ? 'green' : '#000')};
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    z-index: 2;
  }
`;

const StyledTypo = styled(Typography)`
  transition: 0.5s all;
  position: relative;
  z-index: 1;
  opacity: ${({ dragging }) => (dragging ? '0.5' : '1')};
`;

function preventDefaultAction(e) {
  e.preventDefault();
  e.stopPropagation();
}

class DropZone extends React.Component {
  static propTypes = {
    submitFile: PropTypes.func.isRequired,
  }

  constructor(props, context) {
    super(props, context);
    this.state = {
      isDragging: false,
    };
  }


  handleOnDragEnter = (e) => {
    preventDefaultAction(e);
    this.setState({ isDragging: true });
  }

  handleOnDrop = (e) => {
    preventDefaultAction(e);
    const { submitFile } = this.props;
    this.setState({
      isDragging: false,
    });
    const { files } = e.dataTransfer;
    submitFile(files);
  }

  handleOnDragOver = (e) => {
    preventDefaultAction(e);
    this.setState({
      isDragging: true,
    });
    return false;
  }

  handleOnDragLeave = (e) => {
    preventDefaultAction(e);
    this.setState({
      isDragging: false,
    });
  }

  render() {
    const { isDragging } = this.state;
    const text = isDragging ? 'Drop' : 'Drag';
    return (
      <HeightPaper
        dragging={isDragging ? 1 : 0}
        onDragEnter={this.handleOnDragEnter}
        onDragOver={this.handleOnDragOver}
        onDrop={this.handleOnDrop}
        onDragLeave={this.handleOnDragLeave}
      >
        <StyledTypo variant="h5" dragging={isDragging ? 1 : 0}>
          {`${text} files here`}
        </StyledTypo>
      </HeightPaper>
    );
  }
}

export default DropZone;
