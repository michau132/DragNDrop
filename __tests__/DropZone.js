import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DropZone from '../src/DropZone';

const props = {
  submitFile: jest.fn(),
};

describe('<DropZone /> component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<DropZone {...props} />);
  });
  test('should be defined', () => {
    expect(wrapper).toBeDefined();
  });

  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should call submitFile on drop event', () => {
    const prevents = { preventDefault: () => {}, stopPropagation: () => {}, dataTransfer: { files: { name: 'someFile' } } };
    wrapper.simulate('drop', prevents);
    expect(props.submitFile).toHaveBeenCalled();
  });
});
