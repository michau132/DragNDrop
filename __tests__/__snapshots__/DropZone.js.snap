// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropZone /> component should match snapshot 1`] = `
<Styled(WithStyles(Paper))
  dragging={0}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
>
  <Styled(WithStyles(Typography))
    dragging={0}
    variant="h5"
  >
    Drag files here
  </Styled(WithStyles(Typography))>
</Styled(WithStyles(Paper))>
`;
